pkg/config/config.go

```diff
@@ -715,6 +715,10 @@ func ExtractConfigData(yamlData []byte) (common.Configuration, error) {
 		config.S3IngestBufferSize = 1000
 	}
 
+	if config.QueryHostname == "" {
+		config.QueryHostname = fmt.Sprintf("localhost:%v", config.QueryPort)
+	}
```

Comment:
This is pretty similar to `GetQueryServerBaseUrl()`; can we just have one? Or maybe we should keep this but refactor `GetQueryServerBaseUrl()` a little since that's also checking if the hostname is the empty string. If needed, maybe `GetQueryServerBaseUrl()` should have flags for whether to include certain things (like `http/https` and the port)

Reply:
I kept this if check and assigning default. So that `GetQueryHostname()` will return a valid value. And I refactored the `GetQueryServerBaseUrl` and `GetUIDomain` functions to not check for empty `hostname`.